### PR TITLE
Spring Boot SPA article

### PR DESCRIPTION
**Pull request by rtufisi**

### CODE REVIEW

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 1]
```diff
@@ -0,0 +1,278 @@
+---
+slug: secure-spring-boot
+title: Securing SPA Application with Spring Boot and Keycloak
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:3]: Securing **a** SPA Application with Spring Boot and Keycloak

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+author: Phase Two
+tags:
+  [phase_two, tutorial, frameworks, spring-boot, authentication, authorization, sso]
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:6]: `spring_boot` should be snake case.

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+---
+
+import PhaseTwoStarterInstructions from "/templates/blog/_phase_two_starter_instructions.mdx";
+import OIDC from "/templates/blog/_oidc_client_creation_client_auth.mdx";
+import NonAdminUser from "/templates/blog/_add_nonadmin_user.mdx";
+
+[SpringBoot](https://spring.io/projects/spring-boot) is a open-source tool which uses Java based frameworks for building web applications.
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:13]: "Java-based"

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+In this article we'll be using [Keycloak](https://www.keycloak.org/) to secure a SPA and access secured resources from Spring Boot Web application.
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:15]: "from **a** Spring"

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+:::info
+
+If you just want to skip to the code, visit the Phase Two [SpringBoot example](https://github.com/p2-inc/examples/tree/main/frameworks/spring-boot-keycloak). We are also building [Keycloak examples](https://github.com/p2-inc/examples) for other frameworks.
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:19]: correct style is "Spring Boot" with a space.

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+:::
+
+## Setting up a SpringBoot project
+
+In order to setup a SpringBoot project a JDK version must be chosen. To be inline with the latest changes from Keycloak 24, the Java 17 baseline will be used. Other JDk versions can also be used for developing the resource server according to the preference of the developer.
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:25]: "JDK"

File: blog/2024-05-09-secure-spring-boot.mdx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+           );
+           http.csrf(AbstractHttpConfigurer::disable);
+
+           http.sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
+           return http.build();
+       }
+   }
+   ```
+
+This configuration will make the SpringBoot to act as a OAuth2 Resource Server’s  with JWT authentication.
```
Comment (xgp) [inline blog/2024-05-09-secure-spring-boot.mdx:124]: "make the SpringBoot act as a "

"an OAuth2 Resource Server"

**Review by xgp — CHANGES_REQUESTED** [review]
Looks great. A few edits but otherwise :+1:

**Comment by rtufisi** [comment]
@xgp  all changes are performed. Please check again.

Please merge this when you and @jeffpatzer  manage to review this as well. https://github.com/p2-inc/examples/issues/6

Otherwise this section will be incomplete:

`:::info

If you just want to skip to the code, visit the Phase Two [Spring Boot example](https://github.com/p2-inc/examples/tree/main/frameworks/spring-boot-keycloak). We are also building [Keycloak examples](https://github.com/p2-inc/examples) for other frameworks.

:::

`

**Review by xgp — APPROVED** [review]
:+1:

**Comment by xgp** [comment]
@pnzrr looks good from a spring boot, java and keycloak perspective. Please look at the angular part.

**Review by pnzrr — APPROVED** [review]

**Pull request merged**